src/Avalonia.Controls/Primitives/ScrollBar.cs

```diff
@@ -506,5 +513,211 @@ private void UpdatePseudoClasses(Orientation o)
             PseudoClasses.Set(":vertical", o == Orientation.Vertical);
             PseudoClasses.Set(":horizontal", o == Orientation.Horizontal);
         }
+
+        protected override void OnPointerPressed(PointerPressedEventArgs e)
+        {
+            base.OnPointerPressed(e);
+            
+            if (e.GetCurrentPoint(this).Properties.IsRightButtonPressed)
+            {
+                _lastRightClickPosition = e.GetPosition(this);
+            }
+        }
+
+        private void OnContextRequested(object? sender, ContextRequestedEventArgs e)
```

Comment:
Even Microsoft has forgotten the original convention here. But naming with `OnContextRequested` is traditionally reserved for protected virtual methods that can be overridden (like `OnPointerPressed` above). Event handlers have the `Control_ContextRequested` convention -- yes, with an underscore. That is an old tradition I know but it makes total sense in cases like this. You know which one is which immediately.

I think this all started getting convoluted around the time of Xamarin Forms and then Maui.

Reply:
As I said above, I was just trying to map to WPF here. As it's private anyway, I don't think it matters that much but changing it is no cost so if people want to change it, everyone should decide on a name and I'll change it to it.